nifi-stateless/nifi-stateless-api/src/main/java/org/apache/nifi/stateless/config/PropertiesFileEngineConfigurationParser.java

```diff
@@ -111,6 +115,9 @@ public StatelessEngineConfiguration parseEngineConfiguration(final File properti
 
         final String statusTaskInterval = properties.getProperty(STATUS_TASK_INTERVAL, "1 min");
 
+        final long processorStartTimeout = TimeUnit.SECONDS.toMillis(Long.parseLong(properties.getProperty(PROCESSOR_START_TIMEOUT, "10")));
+        final long componentEnableTimeout = TimeUnit.SECONDS.toMillis(Long.parseLong(properties.getProperty(COMPONENT_ENABLE_TIMEOUT, "10")));
```

Comment:
Timeout properties should never be integer values - they need to always be time periods. So "10 secs" is a reasonable default, but we need to parse the property value using `FormatUtils.getTimeDuration`

Reply:
@markap14 After a quick chat with David H. I brought up that FormatUtils is located within `nifi-utils` which is something stateless doesn't already import. What would be your opinion on including this? If the answer is "don't", how would you recommend converting from strings to `Duration`?

Reply:
So it's okay to bring `nifi-utils` into `nifi-stateless-engine`. Definitely not ok to bring it into `nifi-stateless-api` though. So we should probably have the `StatelessEngineConfiguration` return a `String` representation instead of a `Duration`. Then, do the conversion in the `nifi-stateless-engine` module where it'll be used.

It probably actually makes sense to move the FormatUtils functionality into nifi-api because there have been other places where it'd be nice to have it available, such as directly in NiFiProperties. But that's beyond the scope of this Jira :) But something to consider for NiFi 2.0

Reply:
Thanks for the summary @markap14, the FormatUtils has some deprecated methods that need to be removed for NiFi 2.0, and I could see value in pulling it out of `nifi-utils` and into `nifi-api`. In the meantime, it sounds like having the Configuration method return a String value and doing the conversion elsewhere is the best option.